src/asm_writing/types.h

```diff
@@ -53,22 +53,87 @@ struct Register {
     static constexpr int numRegs() { return 16; }
 };
 
-const Register RAX(0);
-const Register RCX(1);
-const Register RDX(2);
-const Register RBX(3);
-const Register RSP(4);
-const Register RBP(5);
-const Register RSI(6);
-const Register RDI(7);
-const Register R8(8);
-const Register R9(9);
-const Register R10(10);
-const Register R11(11);
-const Register R12(12);
-const Register R13(13);
-const Register R14(14);
-const Register R15(15);
+constexpr Register RAX(0);
+constexpr Register RCX(1);
+constexpr Register RDX(2);
+constexpr Register RBX(3);
+constexpr Register RSP(4);
+constexpr Register RBP(5);
+constexpr Register RSI(6);
+constexpr Register RDI(7);
+constexpr Register R8(8);
+constexpr Register R9(9);
+constexpr Register R10(10);
+constexpr Register R11(11);
+constexpr Register R12(12);
+constexpr Register R13(13);
+constexpr Register R14(14);
+constexpr Register R15(15);
+
+struct RegisterSet {
+    typedef unsigned int Regs;
+    Regs regs;
+
+    constexpr explicit RegisterSet(Regs regs) : regs(regs) {}
+    constexpr explicit RegisterSet(Register reg) : regs(1ul << reg.regnum) {}
+
+    static constexpr int max_reg_num = Register::numRegs();
+    static constexpr Regs callee_save_mask = (1ul << RBX.regnum) | (1ul << RSP.regnum) | (1ul << RBP.regnum)
+                                             | (1ul << R12.regnum) | (1ul << R13.regnum) | (1ul << R14.regnum)
+                                             | (1ul << R15.regnum);
+
+    static constexpr RegisterSet std() {
```

Comment:
It might be nice to call this something more descriptive -- "RegisterSet.std" doesn't quite say what it is, so it might be nice to call this something like 'stdAllocatable'?